acir/src/circuit/opcodes.rs

```diff
@@ -6,11 +6,25 @@ use crate::serialization::{read_n, read_u16, read_u32, write_bytes, write_u16, w
 use crate::BlackBoxFunc;
 use serde::{Deserialize, Serialize};
 
+#[derive(Clone, PartialEq, Eq, Serialize, Deserialize, Hash, Copy, Default)]
+pub struct BlockId(u32);
+
+/// Operation on a block
+/// We can either write or read at a block index
+#[derive(Clone, PartialEq, Eq, Serialize, Deserialize)]
+pub struct MemOp {
+    pub operation: Expression,
```

Comment:
As this mirrors the linked review comment in `noir-lang/noir`, I think we should resolve this before merging to ensure that we're treating this consistently.

https://github.com/noir-lang/noir/pull/886/files/2aa2fb5fe16863ca7fabb269dbef601adf9585fa#r1114772704

Reply:
I answered in the PR and resolved the conversation, so I will resolve this one also.